### ISSUE TITLE
manual/Migration22.md

### ISSUE THREAD

**Issue by garbagetown** [playframework-ja/translation-project#861]
- https://github.com/playframework-ja/translation-project/tree/2.3.5/manual/Migration22.md
- diffs between [2.2.0/manual/Migration22.md](https://github.com/playframework-ja/translation-project/tree/2.2.0/manual/Migration22.md) and [2.3.5/manual/Migration22.md](https://github.com/playframework-ja/translation-project/tree/2.3.5/manual/Migration22.md).

```
--- /Users/garbagetown/Desktop/2.2.0/manual/Migration22.md  2013-09-19 10:11:22.000000000 +0900
+++ //Users/garbagetown/Desktop/2.3.5/manual/Migration22.md 2014-07-28 12:26:38.000000000 +0900
@@ -1,6 +1,7 @@
+<!--- Copyright (C) 2009-2013 Typesafe Inc. <http://www.typesafe.com> -->
 # Play 2.2 Migration Guide

-This guide is for migrating a Play 2.1 application to Play 2.2.  To migrate from Play 2.0, first follow the [[Play 2.1 Migration Guide|Migration21]].
+This guide is for migrating to Play 2.2 from Play 2.1.  To migrate to Play 2.1, first follow the [[Play 2.1 Migration Guide|Migration21]].

 ## Build tasks

@@ -48,6 +49,11 @@

 Note that if you depend on plugins that depend on versions of Play prior to 2.2 then there will be a conflict within caching due to multiple caches being loaded. Update to a later plugin version or ensure that older Play versions are excluded if you see this issue.

+### sbt namespace no longer extended
+
+The `sbt` namespace was previously extended by Play e.g. `sbt.PlayCommands.intellijCommandSettings`. This is considered bad practice and so
+Play now uses its own namespace for sbt related things e.g. `play.PlayProject.intellijCommandSettings`.
+
 ## New results structure in Scala

 In order to simplify action composition and filtering, the Play results structure has been simplified.  There is now only one type of result, `SimpleResult`, where before there were `SimpleResult`, `ChunkedResult` and `AsyncResult`, plus the interfaces `Result` and `PlainResult`.  All except `SimpleResult` have been deprecated.  `Status`, a subclass of `SimpleResult`, still exists as a convenience class for building results.  In most cases, actions can still use the deprecated types, but they will get deprecation warnings.  Actions doing composition and filters however will have to switch to using `SimpleResult`.
@@ -112,9 +118,9 @@

 '''java
 public static Promise<Result> myAsyncAction() {
-    Promise<Integer> promiseOfInt = play.libs.Akka.future(
-    new Callable<Integer>() {
-      public Integer call() {
+  Promise<Integer> promiseOfInt = Promise.promise(
+    new Function0<Integer>() {
+      public Integer apply() {
         return intensiveComputation();
       }
     }
@@ -123,7 +129,7 @@
     new Function<Integer, Result>() {
       public Result apply(Integer i) {
         return ok("Got result: " + i);
-      } 
+      }
     }
   );
 }
@@ -190,3 +196,7 @@
 > Please note that the format of the arguments passed to the `start` script has changed. Please issue a `-h` on the `start` script to see the arguments now accepted.

 Please consult the [["Starting your application in production mode"|Production]] documentation for more information on the new `stage` and `dist` tasks.
+
+## Upgrade from Akka 2.1 to 2.2
+
+The migration guide for upgrading from Akka 2.1 to 2.2 can be found [here](http://doc.akka.io/docs/akka/2.2.0/project/migration-guide-2.1.x-2.2.x.html).

```

**Comment by garbagetown** [comment]
:hand:

**Comment by garbagetown** [comment]
merged at #1082
